tools/myst/host/syscall.c

```diff
@@ -16,6 +18,42 @@
         return (ret < 0) ? -errno : ret; \
     } while (0)
 
+#define SAVE_CALL_RESTORE_IDENTITY_RETURN(uid, gid, call)                      \
```

Comment:
If you're going to invoke syscalls anyway, you could make this into a first-class function rather than a macro.

```
int save_call_restore_identity(uid_t uid, gid_t gid, long syscall_number, ...);
```

Reply:
The call to CRT is still there. Only the save/restore IDs are syscalls now